### PR TITLE
Feat/include nsfw channels flag

### PR DESCRIPTION
**Pull request by hozzjss**
# Description
This adds a configuration for the instance maintainer in the discord config file to either include or exclude messages in NSFW channels from cred distribution

Relevant issue #2676 
# Test Plan
unit tests with: yarn test

**Manual test steps**:
run an instance of source cred with discord enabled
I would follow the amazing dev env setup [here](https://github.com/AL0YSI0US/trusty-rabbit-hole/blob/main/setting-up-a-sourcecred-development-env.md)
be sure to have a bot running in a server
add the `includeNsfwChannels` as true in `/config/discord/config.json` so that it might look like this
```json
{
  "guildId": "678348980639498428",
  "reactionWeights": {
    "👍": 1,
    "sourcecred:678399364418502669": 10
  },
  "includeNsfwChannels": true
}
```
Check if cred from nsfw channels is being calculated after running
`scdev go` and then `scdev serve` and checking the explorer

then change `includeNsfwChannels` to false or remove it entirely and check if cred is not distributed to senders in that channels based on reactions and messages.

### CODE REVIEW

**Review by AL0YSI0US — APPROVED** [review]
# 👨🏾‍💻 @hozzjss . . . 
I really dug the Test plan you wrote up.
<div align="center">

It was so helpful to read before looking at the changes that were made.

**Appreciating you.**

Also.....what an epic pull request number 🤘🏾 </div>

````                                                               
  .x~~"*Weu.       .n~~%x.         .n~~%x.         .n~~%x.     
 d8Nu.  9888c    x88X   888.     x88X   888.     x88X   888.            🏆
 88888  98888   X888X   8888L   X888X   8888L   X888X   8888L        
 "***"  9888%  X8888X   88888  X8888X   88888  X8888X   88888         N E W
      ..@8*"   88888X   88888X 88888X   88888X 88888X   88888X      
   ````"8Weu   88888X   88888X 88888X   88888X 88888X   88888X        H I G H   
  ..    ?8888L 88888X   88888f 88888X   88888f 88888X   88888f   
:@88N   '8888N 48888X   88888  48888X   88888  48888X   88888        S C O R E
*8888~  '8888F  ?888X   8888"   ?888X   8888"   ?888X   8888"  
'*8"`   9888%    "88X   88*`     "88X   88*`     "88X   88*`           ! ! !
  `~===*%"`        ^"==="`         ^"==="`         ^"==="`     
                                                               
````

**Comment by hozzjss** [comment]
Thanks @AL0YSI0US absolutely, and let me know if you need credentials or anything if you're blocked from doing the manual testing

**Comment by topocount** [comment]
Hey @hozzjss you've been invited to the org, and don't have to push from forks anymore

**Comment by hozzjss** [comment]
image

**Review by topocount — COMMENTED** [review]
Hey Hozz this implementation is looking good. My only request is that we have separate tests that explicitly show that this option works, possibly by loading nsfw channel(s) in a snapshot when enabled, and not loading them into a separate snapshot when disabled.

**Comment by hozzjss** [comment]
That makes sense, I'll work on that

File: src/plugins/discord/config.test.js [hunk 1]
```diff
@@ -42,5 +42,6 @@ describe("plugins/discord/config", () => {
       weights: {},
       defaultWeight: 1,
     });
+    expect(parsed.includeNsfwChannels).toEqual(false);
   });
 });
```
Comment (blueridger) [inline src/plugins/discord/config.test.js:47]: Would you add the property to the first test case as "true" to test the true case as well

Comment (hozzjss) [reply src/plugins/discord/config.test.js:47]: I'll do that

Comment (hozzjss) [reply src/plugins/discord/config.test.js:49]: and done

**Comment by hozzjss** [comment]
Just added tests and updated the snapshots

**Review by blueridger — APPROVED** [review]
Really impressive work! This had some complexity to it, and completing it so independently for your 2nd task really shows your skill.

Pulled this code into a branch in our repo to make sure full test suite passes.

**Pull request merged**